flink-tests/src/test/java/org/apache/flink/test/runtime/HybridShuffleITCase.java

```diff
@@ -0,0 +1,203 @@
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
+import org.apache.flink.streaming.api.functions.source.ParallelSourceFunction;
+import org.apache.flink.streaming.api.graph.StreamGraph;
+import org.apache.flink.streaming.api.graph.StreamingJobGraphGenerator;
+import org.apache.flink.streaming.api.operators.StreamSource;
+import org.apache.flink.testutils.junit.utils.TempDirUtils;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Tests for hybrid shuffle mode. */
+class HybridShuffleITCase {
+    private static final String RECORD = "hybrid shuffle test";
+
+    private static final int NUM_TASK_MANAGERS = 2;
+
+    private static final int NUM_SLOTS_PER_TASK_MANAGER = 10;
+
+    private static final int PARALLELISM = NUM_SLOTS_PER_TASK_MANAGER;
+
+    private static final int[] NUM_RECEIVED_RECORDS = new int[PARALLELISM];
+
+    private static Path tmpDir;
+
+    @BeforeAll
+    static void setup(@TempDir Path path) throws Exception {
+        tmpDir = TempDirUtils.newFolder(path, UUID.randomUUID().toString()).toPath();
+    }
+
+    @Test
+    void testHybridFullExchanges() throws Exception {
+        final int numRecordsToSend = 10000;
+        Configuration configuration = getConfiguration();
+        configuration.set(
+                ExecutionOptions.BATCH_SHUFFLE_MODE, BatchShuffleMode.ALL_EXCHANGES_HYBRID_FULL);
+        JobGraph jobGraph = createJobGraph(numRecordsToSend, false, configuration);
+        JobGraphRunningUtil.execute(
+                jobGraph, configuration, NUM_TASK_MANAGERS, NUM_SLOTS_PER_TASK_MANAGER);
+        checkAllDataReceived(numRecordsToSend);
+    }
+
+    @Test
+    void testHybridSelectiveExchanges() throws Exception {
+        final int numRecordsToSend = 10000;
+        Configuration configuration = getConfiguration();
+        configuration.set(
+                ExecutionOptions.BATCH_SHUFFLE_MODE,
+                BatchShuffleMode.ALL_EXCHANGES_HYBRID_SELECTIVE);
+        JobGraph jobGraph = createJobGraph(numRecordsToSend, false, configuration);
+        JobGraphRunningUtil.execute(
+                jobGraph, configuration, NUM_TASK_MANAGERS, NUM_SLOTS_PER_TASK_MANAGER);
+        checkAllDataReceived(numRecordsToSend);
+    }
+
+    @Test
+    void testHybridFullExchangesRestart() throws Exception {
+        final int numRecordsToSend = 10;
+        Configuration configuration = getConfiguration();
+        configuration.set(
+                ExecutionOptions.BATCH_SHUFFLE_MODE, BatchShuffleMode.ALL_EXCHANGES_HYBRID_FULL);
+        configuration.set(NettyShuffleEnvironmentOptions.NETWORK_REQUEST_BACKOFF_MAX, 100);
```

Comment:
We can move this to `getConfiguration()`.

Reply:
Moved.